tests/markov/msm/test_msm.py

```diff
@@ -242,797 +220,485 @@ def test_valid_trajectory(self):
+    assert 1.0 <= s1 <= 2.0
+    s2 = score_cv(fit_fetch, dtrajs=scenario.dtraj, lagtime=10, n=5, score_method='VAMP2', score_k=2).mean()
+    assert 1.0 <= s2 <= 2.0
+    se = score_cv(fit_fetch, dtrajs=scenario.dtraj, lagtime=10, n=5, score_method='VAMPE', score_k=2).mean()
+    se_inf = score_cv(fit_fetch, dtrajs=scenario.dtraj, lagtime=10, n=5, score_method='VAMPE', score_k=None).mean()
+
+
+def test_reversible_property(double_well_msm_all):
+    _, est, msm = double_well_msm_all
+    assert_equal(est.reversible, msm.reversible)
+
+
+def test_sparse_property(double_well_msm_all):
+    _, est, msm = double_well_msm_all
+    assert_equal(est.sparse, msm.sparse)
+
+
+def test_lagtime_property(double_well_msm_all):
+    scenario, est, msm = double_well_msm_all
+    assert_equal(msm.lagtime, scenario.lagtime)
+
+
+def test_state_symbols(double_well_msm_all):
+    scenario, est, msm = double_well_msm_all
+    # should always be <= full set
+    assert_(len(msm.count_model.state_symbols) <= msm.count_model.n_states_full)
+    # should be length of n_states
+    assert_equal(len(msm.count_model.state_symbols), msm.count_model.n_states)
+
+
+def test_n_states_property(double_well_msm_all):
+    scenario, est, msm = double_well_msm_all
+    assert_(msm.n_states <= msm.count_model.n_states)
+    assert_equal(msm.n_states, scenario.n_states)
+
+
+def test_connected_sets(double_well_msm_all):
+    scenario, est, msm = double_well_msm_all
+    cs = msm.count_model.connected_sets()
+    assert_equal(len(cs), 1)
+    # mode largest: re-evaluating connected_sets should yield one connected set with exactly as many states as
+    # contained in the count model
+    assert_equal(cs[0], np.arange(msm.count_model.n_states))
+
+
+@pytest.mark.parametrize("full", [False, True], ids=["not full", "full"])
+def test_count_matrix(double_well_msm_all, full):
+    scenario, est, msm = double_well_msm_all
+    if full:
+        count_matrix = msm.count_model.count_matrix_full
+        n = np.max(scenario.dtraj) + 1
+        assert_equal(count_matrix.shape, (n, n))
+    else:
+        count_matrix = msm.count_model.count_matrix
+        assert_equal(count_matrix.shape, (msm.n_states, msm.n_states))
+
+
+def test_discrete_trajectories_active(double_well_msm_all):
+    scenario, est, msm = double_well_msm_all
+    dta = msm.count_model.transform_discrete_trajectories_to_submodel(scenario.dtraj)
+    assert_equal(len(dta), 1)
+    # HERE: states are shifted down from the beginning, because early states are missing
+    assert_(dta[0][0] < scenario.dtraj[0])
+
+
+def test_physical_time(double_well_msm_all):
+    _, _, msm = double_well_msm_all
+    assert_(str(msm.count_model.physical_time).startswith('1'))
+    assert_(str(msm.count_model.physical_time).endswith('step'))
+
+
+def test_transition_matrix(double_well_msm_all):
+    scenario, est, msm = double_well_msm_all
+    P = msm.transition_matrix
```

Comment:
if you feel like it you can change this `P` too but not essential

Reply:
eh yeah thats just in a test :man_shrugging: